pkg/cluster/sync.go

```diff
@@ -332,18 +332,6 @@ func (c *Cluster) syncStatefulSet() error {
 		// statefulset is already there, make sure we use its definition in order to compare with the spec.
 		c.Statefulset = sset
 
-		// check if there is no Postgres version mismatch
-		for _, container := range c.Statefulset.Spec.Template.Spec.Containers {
```

Comment:
Is this code obsolete?

Reply:
This was initially done [to prevent breaking the cluster by altering the PG version](https://github.com/zalando/postgres-operator/pull/860).  With this PR, increasing a PG version becomes a legitimate operation; decreasing a PG version is still caught with this PR.